tls/s2n_handshake_io.c

```diff
@@ -455,6 +530,10 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
         conn->handshake.handshake_type |= CLIENT_AUTH;
     }
 
+    if (IS_TLS13_HANDSHAKE(conn)) {
```

Comment:
Should this be after the `s2n_kex_is_ephemeral()` check below? Otherwise the `PERFECT_FORWARD_SECRECY` bit will never be set for TLS 1.3 handshakes.

Reply:
That's intentional. The PERFECT_FORWARD_SECRECY flag (which just meant that we sent the SERVER_KEY message in TLS1.2) isn't relevant in TLS1.3, where all cipher suites support perfect forward secrecy and key material is exchanged in extensions.

Reply:
Hmm, then I think we should rename the `PERFECT_FORWARD_SECRECY` flag to `LEGACY_PERFECT_FORWARD_SECRECY` or `TLS_12_PERFECT_FORWARD_SECRECY` to clarify.